src/redis_ai_objects/tensor.c

```diff
@@ -24,6 +25,23 @@
 
 extern RedisModuleType *RedisAI_TensorType;
 
+// Check if the given value is in the range of the tensor type.
+bool _ValOverflow(long long val, RAI_Tensor *t) {
+    DLDataType dtype = t->tensor.dl_tensor.dtype;
+    if (dtype.code == kDLInt) {
+        uint max_abs_val = (uint)1 << (uint)(dtype.bits - 1);
+        if (val >= (long long)max_abs_val || val <= (long long)-1 * max_abs_val) {
+            return true;
+        }
+    } else if (dtype.code == kDLUInt) {
+        uint max_val = (uint)1 << dtype.bits;
+        if (val >= max_val || val < 0) {
+            return true;
```

Comment:
please add a comment that we support up to 32 bits of unsigned int
also, please keep all the types in this functions as `long long`